### PR TITLE
Made http_redirect_fetch error for non-HTTPS. #14069

### LINKED ISSUE #14069: Make HTTP redirect fetch return an error if redirecting to non-HTTP(S)
Step 4 of the [http redirect fetch](https://fetch.spec.whatwg.org/#concept-http-redirect-fetch) specification says that we should return an error if we try to redirect to a non-HTTP(S) URL. Our corresponding implementation has a TODO in `http_redirect_fetch`. We should check the [scheme](http://doc.servo.org/url/struct.Url.html#method.scheme) of `location_url` and return `Response::network_error` if it doesn't match.

Code: `components/net/fetch/methods.rs`
Test:
`./mach test-wpt tests/wpt/web-platform-tests/fetch/api/redirect/redirect-schemes.html` should pass after this change. We should remove [the test expectation file](https://dxr.mozilla.org/servo/source/tests/wpt/metadata/fetch/api/redirect/redirect-schemes.html.ini) in that case.

### PR DESCRIPTION
**Pull request by 46bit**
Hi! I'm a newbie looking to resolve #14069.

The [existing tests](https://dxr.mozilla.org/servo/source/tests/wpt/web-platform-tests/fetch/api/redirect/redirect-schemes.html) now pass. 

Cheers!

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #14069 (github issue number if applicable).

- [X] There are tests for these changes OR

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14420)

### CODE REVIEW

File: components/net/http_loader.rs [hunk 1]
```diff
@@ -772,7 +772,11 @@ fn http_redirect_fetch(request: Rc<Request>,
     };
 
     // Step 4
-    // TODO implement return network_error if not HTTP(S)
+    match location_url.scheme() {
+        "http" => { },
+        "https" => { },
```
Comment (KiChjang) [inline components/net/http_loader.rs]: This can just be `"http" | "https" => { }`.

File: components/net/http_loader.rs [hunk 2]
```diff
@@ -772,7 +772,11 @@ fn http_redirect_fetch(request: Rc<Request>,
...
+        _ => return Response::network_error(NetworkError::Internal("Not an HTTPS Scheme".into()))
```
Comment (KiChjang) [inline components/net/http_loader.rs]: `"Not an HTTP(S) scheme"`.

**Review by KiChjang — CHANGES_REQUESTED** [review]
Thanks for your PR! There are a couple of minor changes, r=me after they're fixed.

For the questions you posed, I don't think it's necessary to resolve them in this PR since they're out of scope. I'd also encourage you to comment on the relevant issue first before working on it next time.

**Comment by 46bit** [comment]
r=KiChjang

**Comment by KiChjang** [comment]
@46bit That message was intended for other reviewers indicating that this patch can be landed if I happen to be away for a while. Thanks for your contribution nonetheless!

@bors-servo r+

**Review by KiChjang — APPROVED** [review]

**Comment by KiChjang** [comment]
Looks like more tests are passing after your changes:
```
Tests with unexpected results:
  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl.html:
  └ PASS [expected FAIL] Testing data URL loading after same-origin redirection (cors mode)

  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl.html:
  └ PASS [expected FAIL] Testing data URL loading after same-origin redirection (no-cors mode)

  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl.html:
  └ PASS [expected FAIL] Testing data URL loading after same-origin redirection (same-origin mode)

  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl.html:
  └ PASS [expected FAIL] Testing data URL loading after cross-origin redirection (cors mode)

  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl.html:
  └ PASS [expected FAIL] Testing data URL loading after cross-origin redirection (no-cors mode)

  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl-worker.html:
  └ PASS [expected FAIL] Testing data URL loading after same-origin redirection (cors mode)

  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl-worker.html:
  └ PASS [expected FAIL] Testing data URL loading after same-origin redirection (no-cors mode)

  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl-worker.html:
  └ PASS [expected FAIL] Testing data URL loading after same-origin redirection (same-origin mode)

  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl-worker.html:
  └ PASS [expected FAIL] Testing data URL loading after cross-origin redirection (cors mode)

  ▶ Unexpected subtest result in /fetch/api/redirect/redirect-to-dataurl-worker.html:
  └ PASS [expected FAIL] Testing data URL loading after cross-origin redirection (no-cors mode)
```
You'll need to update the test expectations accordingly.

**Comment by 46bit** [comment]
Aha. I've removed the outdated expectations and checked the full `./mach test-wpt tests/wpt/web-platform-tests/fetch`.

**Comment by KiChjang** [comment]
Thanks!

@bors-servo r+

**Pull request merged**